### ISSUE TITLE
Future of Coding Weekly 2023/11 Week 1

### ISSUE THREAD

**Issue by marianoguerra** [marianoguerra/future-of-coding-weekly#216]
- [x] Change Month and Week Number
- [x] focWeekExport "2023-01-19" "2023-01-26"
- [x] Update Search Index
- [x] Download New Attachments
- [x] Update links
- [x] Check that comment links work (push weekly dump with channel summaries)
- [ ] Check to mention right person for moved messages
- [ ] Summary
- [ ] Hashtags
- [ ] Set title in newsletter

- https://tinyletter.com/
- https://tinyletter.com/marianoguerra/letters/
- http://localhost:8000/history/
- https://marianoguerra.github.io/future-of-coding-weekly/
- https://stackedit.io/app#

**Comment by marianoguerra** [comment]
🎧 Future of Scrappy Music ⌨️ Wordplay Beta 🎥 Demos with Squares, Disks & Loops

# Two Minute Week

🎥 [Debug Coroutine Demo](http://history.futureofcoding.org/history/msg_files/F06/F063QU4LCSE.webm) via [Kartik Agaram](http://akkartik.name/about)

[🧵 conversation](https://history.futureofcoding.org/history/weekly/2023/11/W1/two-minute-week.html#2023-11-01T06:07:08.191Z)

I've been experimenting with ways to debug my live programs.



(Kind of a follow-up to [archive.org/details/akkartik-mu-2021-05-17](https://archive.org/details/akkartik-mu-2021-05-17), if you squint.)

🎥 [Debug Coroutine Demo](http://history.futureofcoding.org/history/msg_files/F06/F063QU4LCSE.webm)

**Comment by marianoguerra** [comment]
🎥 [Projected AR](http://history.futureofcoding.org/history/msg_files/F06/F063QPFPKDK.mp4) by **Arcade Wise**

[🧵 conversation](https://history.futureofcoding.org/history/weekly/2023/11/W1/two-minute-week.html#2023-11-01T15:54:22.446Z)

Not quite 2 minutes, but a fun little set-up for messing around with projected AR concepts!

🎥 [Projected AR](http://history.futureofcoding.org/history/msg_files/F06/F063QPFPKDK.mp4)

**Comment by marianoguerra** [comment]
🎥 [Rectangle World](http://history.futureofcoding.org/history/msg_files/F06/F063WUUVB2A.webm) by [Kartik Agaram](http://akkartik.name/about)

[🧵 conversation](https://history.futureofcoding.org/history/weekly/2023/11/W1/two-minute-week.html#2023-11-03T05:43:33.638Z)

Rectangle World, a little test bed for exploring easy versions of some problems, along with techniques for writing easy-to-debug programs.

🎥 [Rectangle World](http://history.futureofcoding.org/history/msg_files/F06/F063WUUVB2A.webm)

**Comment by marianoguerra** [comment]
# Our Work

⌨️ [Wordplay beta](https://wordplay.dev) via **Amy Ko**

[🧵 conversation](https://history.futureofcoding.org/history/weekly/2023/11/W1/share-your-work.html#2023-10-31T15:06:38.778Z)

[Wordplay beta](https://wordplay.dev) is live! We’d love your feedback on everything (language design, APIs, tools, editors), and your contributions, if you’re so inclined.



- [Blog](https://medium.com/bits-and-behavior/wordplay-accessible-language-inclusive-interactive-typography-e4b9027eaf10)
- [Repo](https://github.com/wordplaydev/wordplay)
- [Discord](https://discord.gg/Jh2Qq9husy)

**Comment by marianoguerra** [comment]
🎨 [MightyMeld: a future kind of dev tool](https://www.mightymeld.com/) via **Steven Schkolne**

[🧵 conversation](https://history.futureofcoding.org/history/weekly/2023/11/W1/share-your-work.html#2023-11-01T19:43:40.599Z)

Hey everyone -- long-time lurker, first time poster. Thrilled to share the open beta launch of MightyMeld, a visual dev tool for React I've been building with some friends of mine. [mightymeld.com](https://www.mightymeld.com/)



Just posted this blog post telling the story and why I'm excited -- [mightymeld.com/blog/open-beta](https://mightymeld.com/blog/open-beta)



Would love to hear your thoughts! I'm a former academic who's been driven into industry out of a desire to make the theoretical practical. MightyMeld is a "visual coding" that is not like any kind of "visual coding" i would've imagined if we hadn't've built this thing starting from real production code and saying "how can we make this more visual".

**Comment by marianoguerra** [comment]
📝 [From app trap to freedom space](https://duncancragg.substack.com/p/from-app-trap-to-freedom-space) via [Duncan Cragg](https://twitter.com/duncancragg)

[🧵 conversation](https://history.futureofcoding.org/history/weekly/2023/11/W1/share-your-work.html#2023-11-02T11:54:34.657Z)

Hiya, if you're interested in:






* End-User Programming, Declarative languages
* Digital Gardens or Personal Knowledge Managers/Graphers
* Decentralisation



Then you clearly need to read my latest fine work: [From app trap to freedom space](https://duncancragg.substack.com/p/from-app-trap-to-freedom-space)



TL;DR: we can break free of the "app trap" by simply building an  _OS without apps_ !



(Don't forget to subscribe so you won't miss future updates, right into your mailbox...)

**Comment by marianoguerra** [comment]
🎧 [arroost.com](http://arroost.com) via [Lu Wilson](https://twitter.com/TodePond)

[🧵 conversation](https://history.futureofcoding.org/history/weekly/2023/11/W1/share-your-work.html#2023-11-03T18:35:14.363Z)

Hello everyone I've been making a new tool called arroost. It's for making scrappy music. Please do make something in it and send it to me.




I will post some examples of its use in the replies

**Comment by marianoguerra** [comment]
# Devlog Together

📝 [Experimenting with chatgpt as my ghost writer](https://guitarvydas.github.io/2023/10/29/Experimenting-With-ChatGPT-As-My-Ghost-Writer.html) via [Paul Tarvydas](https://guitarvydas.github.io/2021/09/23/Manifesto.html)

[🧵 conversation](https://history.futureofcoding.org/history/weekly/2023/11/W1/devlog-together.html#2023-11-01T01:35:54.703Z)

I’m developing a cutting-edge tool that leverages ChatGPT to ghost-write prose based on my point-form outlines, particularly focusing on ‘new ideas’ that the AI wasn’t trained on. The challenge lies in preventing hallucinations and incorrect extrapolations from the outline. Early experimental results have been promising prompting me to create a dedicated workflow tool. This tool enables me to swiftly input point-form outlines, using the user-friendly Kinopio platform for brainstorming. I then paste the outline to ChatGPT, transforming it into coherent and readable prose. Join me in revolutionizing idea development and content creation with this innovative solution! [ChatGPT]

**Comment by marianoguerra** [comment]
🎥 [Rectangle World II](http://history.futureofcoding.org/history/msg_files/F06/F0648SKJLPM.webm) via [Kartik Agaram](http://akkartik.name/about)

[🧵 conversation](https://history.futureofcoding.org/history/weekly/2023/11/W1/devlog-together.html#2023-11-04T05:25:11.054Z)

Today's follow-up to [💬 #two-minute-week@2023-11-03](./?fromDate=2023-10-31&toDate=2023-11-06&channel=two-minute-week&filter=#2023-11-03T05:43:33.638Z) as I try to come up with more pleasing random arrangements in Rectangle World.

**Comment by marianoguerra** [comment]
🎥 [Poisson Disk Sampling Debug UI](http://history.futureofcoding.org/history/msg_files/F06/F0647F7ERFF.webm) via [Kartik Agaram](http://akkartik.name/about)

[🧵 conversation](https://history.futureofcoding.org/history/weekly/2023/11/W1/devlog-together.html#2023-11-05T04:52:05.534Z)

Some satisfying debug UI for Poisson Disk Sampling ([sighack.com/post/poisson-disk-sampling-bridsons-algorithm](https://sighack.com/post/poisson-disk-sampling-bridsons-algorithm); thanks [Jack Rusher](https://twitter.com/jackrusher) for pointing me at it)



Couple more videos in 🧵

🎥 [Poisson Disk Sampling Debug UI](http://history.futureofcoding.org/history/msg_files/F06/F0647F7ERFF.webm)

**Comment by marianoguerra** [comment]
🎥 [Creating new flow using a "back-loop" and placing the "back-loop" connection on a separate layer](https://youtu.be/cnD-AlDQJDk) via [Maikel van de Lisdonk](https://www.devhelpr.com/)

[🧵 conversation](https://history.futureofcoding.org/history/weekly/2023/11/W1/devlog-together.html#2023-11-05T13:37:58.844Z)

![Youtube Thumbnail](https://img.youtube.com/vi/cnD-AlDQJDk/hqdefault.jpg)

Hi, the last month I've been focussing a lot on fixing some bugs in the way nodes are connected and improving the different path types that are used.



This small demo video [(Creating new flow using a "back-loop" and placing the "back-loop" connection on a separate layer)](https://youtu.be/cnD-AlDQJDk) shows creating a very small flow from scratch, the functionality of the flow is not important here.. but the thing I wanted to show is the way a "backloop" is created. The system uses a different svg path for backloops: normal nodes are connected using a bezier curve but when the output node is placed before the start node, the path changes to a "back-loop". 



Another thing to notice is the layering.. the back-loop is placed on its own layer, hopefully keeping bigger flows more clear. You can toggle which layer is in front.

**Comment by marianoguerra** [comment]
# Thinking Together

🧑‍🎨 [music thing](https://www.figma.com/proto/rOhbNvD6U1MEkM3zeznLeJ/music-thing?page-id=0%3A1&type=design&node-id=2-3&viewport=436%2C-512%2C0.9&t=mYfeXCUoQZiP9aDh-1&scaling=contain&starting-point-node-id=2%3A3&mode=design) via **Arcade Wise**

[🧵 conversation](https://history.futureofcoding.org/history/weekly/2023/11/W1/thinking-together.html#2023-10-30T14:55:56.638Z)

I've been working on learning UI design, what do y'all think of this?  [music thing](https://www.figma.com/proto/rOhbNvD6U1MEkM3zeznLeJ/music-thing?page-id=0%3A1&type=design&node-id=2-3&viewport=436%2C-512%2C0.9&t=mYfeXCUoQZiP9aDh-1&scaling=contain&starting-point-node-id=2%3A3&mode=design)

**Comment by marianoguerra** [comment]
💬 **John Flockton**

[🧵 conversation](https://history.futureofcoding.org/history/weekly/2023/11/W1/thinking-together.html#2023-11-02T17:27:38.130Z)

Hey folks! I'm doing some research into code formatters/pretty printers, with a view to potentially building my own. Does any one have any examples of code formatters that they've found especially fast, or that they know of which have a special focus on performance?



I'm interested in learning about some of the potential techniques at play - what sort of caching and parallelization is being done, are they doing incremental parsing like tree-sitter etc..

**Comment by marianoguerra** [comment]
💬 **Mike Austin**

[🧵 conversation](https://history.futureofcoding.org/history/weekly/2023/11/W1/thinking-together.html#2023-11-02T19:37:50.722Z)

Over the years, I've thought about different ways of creating an interactive tutorial about writing a simple programming language. For example, you can create a "wizard" style with next and back buttons, or you can create one single page with the code floating on the right side of the content (possibly sticky). One detail is that each example would grow in size (for the complete code), but would only focus on one sub-section. I thought about collapsing sections already covered, but being able to expand them, or just have a single button to expand all the hidden code. Has anyone had experience or have seen articles that use a "build from small to large", showing code along the way?

**Comment by marianoguerra** [comment]
💬 **Arcade Wise**

[🧵 conversation](https://history.futureofcoding.org/history/weekly/2023/11/W1/linking-together.html#2023-11-04T00:48:42.654Z)

This is a cute little section from the wordmaster manual :>


![image.png](http://history.futureofcoding.org/history/msg_files/F06/F064MA50997.png)

I did in fact get the 12 lines

![image.png](http://history.futureofcoding.org/history/msg_files/F06/F063U53HMGF.png)

considered harmful?

![image.png](http://history.futureofcoding.org/history/msg_files/F06/F0648K1M0KV.png)

**Comment by marianoguerra** [comment]
# 🤖

💬 **Nicolay Gerold**

[🧵 conversation](https://history.futureofcoding.org/history/weekly/2023/11/W1/of-ai.html#2023-10-30T11:27:49.861Z)

Hey 🙂



I am looking for inspiration in building frontends with LLMs. I am looking for companies, which have LLMs (their own or wrappers around GPT) deployed in customer-facing applications (B2B and B2C) at some level of scale.



Something like Sudowrite and Jasper.



Also if you know some impressive solobuilders, feel free to name drop them 🙂



I would appreciate any link or name dropped in the thread 😉
